### PR TITLE
Remove use of front endpoint in poke

### PR DESCRIPTION
**Pull request by spolu**
## Description

Goals is to properly enforce everywhere under `/w/[wId]/` the use of `withSessionAuthenticationForWorkspace`

Fixes: https://github.com/dust-tt/tasks/issues/1205

r? @flvndvd 
```
- // TODO(2024-07-09 flav) Create a dedicated endpoint on Poke.
```

## Risk

Auth: double check poke logic

## Deploy Plan

- deploy `front`

### CODE REVIEW

File: front/lib/swr.ts [hunk 1]
```diff
@@ -627,6 +624,29 @@ export function usePokeConnectorPermissions({
   };
 }
 
+export function usePokeDocuments(
```
Comment (flvndvd) [inline front/lib/swr.ts:627]: We introduced poke/swr can you move it there?

File: front/pages/api/poke/workspaces/[wId]/data_sources/[name]/documents/index.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+};
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<WithAPIErrorResponse<GetDocumentsResponseBody>>
+): Promise<void> {
+  const session = await getSession(req, res);
+  const auth = await Authenticator.fromSuperUserSession(
+    session,
+    req.query.wId as string
```
Comment (flvndvd) [inline front/pages/api/poke/workspaces/[wId]/data_sources/[name]/documents/index.ts:24]: `nit`

Let's avoid casting and do:

```
const {wId} = req.query;

if (typeof wId !== "string") {
  // Handle error.
} 
```

File: front/pages/api/poke/workspaces/[wId]/data_sources/[name]/documents/index.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "user_not_found",
+        message: "Could not find the user.",
+      },
+    });
+  }
+
+  const dataSource = await getDataSource(auth, req.query.name as string);
```
Comment (flvndvd) [inline front/pages/api/poke/workspaces/[wId]/data_sources/[name]/documents/index.ts:38]: Same for name.

**Review by flvndvd — APPROVED** [review]
🙏

**Comment by spolu** [comment]
(updated :+1:)

**Review by flvndvd — APPROVED** [review]

**Pull request merged**